src/Framework/test/PackageTests.cs

```diff
@@ -0,0 +1,139 @@
+            // Don't test helix test runner tool packages
+            if (helixTestRunnerToolPackages.Any(s => packageDir.Contains(s, StringComparison.OrdinalIgnoreCase)))
+            {
+                continue;
+            }
+            // Test lib assemblies
+            var packageAssembliesDir = Path.Combine(packageDir, "lib");
+            if (Directory.Exists(packageAssembliesDir))
+            {
+                foreach (var tfmDir in Directory.GetDirectories(packageAssembliesDir))
+                {
+                    var tfm = new DirectoryInfo(tfmDir).Name;
+                    foreach (var assembly in Directory.GetFiles(tfmDir, "*.dll"))
+                    {
+                        using var fileStream = File.OpenRead(assembly);
+                        using var peReader = new PEReader(fileStream, PEStreamOptions.Default);
+                        var reader = peReader.GetMetadataReader(MetadataReaderOptions.Default);
+                        var assemblyVersion = reader.GetAssemblyDefinition().Version;
+
+                        // net & netstandard assembly versions should all match Major.Minor.0.0
+                        // netfx assembly versions should match Major.Minor.Patch.0
+                        Assert.Equal(expectedVersion.Major, assemblyVersion.Major);
+                        Assert.Equal(expectedVersion.Minor, assemblyVersion.Minor);
+                        if (IsNetFx(tfm))
+                        {
+                            Assert.Equal(expectedVersion.Build, assemblyVersion.Build);
+                        }
+                        else
+                        {
+                            Assert.Equal(0, assemblyVersion.Build);
+                        }
+                        Assert.Equal(0, assemblyVersion.Revision);
+                    }
+                }
+            }
+
+            // Test tool assemblies
+            var packageToolsDir = Path.Combine(packageDir, "tools");
+            if (Directory.Exists(packageToolsDir))
+            {
+                var assemblies = Directory.GetFiles(packageToolsDir, "*.dll", SearchOption.AllDirectories)
+                    .Where(f => !toolAssembliesToSkip.Any(s => Path.GetFileNameWithoutExtension(f).Contains(s, StringComparison.OrdinalIgnoreCase)));
+                foreach (var assembly in assemblies)
+                {
+                    using var fileStream = File.OpenRead(assembly);
+                    using var peReader = new PEReader(fileStream, PEStreamOptions.Default);
+                    var reader = peReader.GetMetadataReader(MetadataReaderOptions.Default);
+                    var assemblyVersion = reader.GetAssemblyDefinition().Version;
+
+                    Assert.Equal(expectedVersion.Major, assemblyVersion.Major);
+                    Assert.Equal(expectedVersion.Minor, assemblyVersion.Minor);
+                    Assert.Equal(0, assemblyVersion.Build);
+                    Assert.Equal(0, assemblyVersion.Revision);
```

Comment:
nit: This is a bit repetitive. Suggest passing `assembly` and an `expectedVersion` (that's already handled the .NET Framework choice) into a method.

Separately, we don't ship any .NET Framework tool assemblies❔

Reply:
We don't ship any netfx tool assemblies today. 

We check assemblyVersions the same way in SharedFxTests & TargetingPackTests, I don't feel like fixing all 3 in this PR - maybe as a follow up

Reply:
Sounds fine.

Reply:
Did you file a follow up (nit) issue❔

Reply:
I don't think this is important enough to have an issue for (or even to proactively fix, if I'm being honest)